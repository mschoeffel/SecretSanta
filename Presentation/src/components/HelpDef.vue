<template>
  <v-container class="fill-height" fluid>
    <v-row align="center" justify="center">
      <v-col cols="12" sm="12" md="8">
        <v-card class="elevation-12">
          <v-toolbar color="primary" dark flat>
            <v-toolbar-title name="title" id="title">{{ $t('help.formtitle')}}</v-toolbar-title>
          </v-toolbar>
          <v-card-text>
            <v-expansion-panels>
              <v-expansion-panel>
                <v-expansion-panel-header>{{$t('help.firststeps')}}</v-expansion-panel-header>
                <v-expansion-panel-content>{{$t('help.firststeps-content')}}</v-expansion-panel-content>
              </v-expansion-panel>
              <v-expansion-panel>
                <v-expansion-panel-header>{{$t('help.newgroup')}}</v-expansion-panel-header>
                <v-expansion-panel-content>{{$t('help.newgroup-content')}}</v-expansion-panel-content>
              </v-expansion-panel>
              <v-expansion-panel>
                <v-expansion-panel-header>{{$t('help.showgroup')}}</v-expansion-panel-header>
                <v-expansion-panel-content>{{$t('help.showgroup-content')}}</v-expansion-panel-content>
              </v-expansion-panel>
              <v-expansion-panel>
                <v-expansion-panel-header>{{$t('help.drawmember')}}</v-expansion-panel-header>
                <v-expansion-panel-content>{{$t('help.drawmember-content')}}</v-expansion-panel-content>
              </v-expansion-panel>
            </v-expansion-panels>
          </v-card-text>
          <v-card-actions>
            <v-btn
              name="routehome"
              id="routehome"
              color="secondary"
              v-on:click="routeHome"
            >{{ $t('help.home')}}</v-btn>
          </v-card-actions>
        </v-card>
      </v-col>
    </v-row>
  </v-container>
</template>

<script>
export default {
  props: {
    source: String
  },
  data: () => ({}),
  methods: {
    routeHome: function() {
      this.$router.push({ path: "/" });
    }
  }
};
</script>