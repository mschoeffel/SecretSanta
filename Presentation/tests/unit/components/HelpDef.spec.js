import Vue from 'vue'
import Vuetify from 'vuetify'
import VueI18n from 'vue-i18n'

import HelpDef from '@/components/HelpDef'

import { mount, createLocalVue } from '@vue/test-utils';

const localVue = createLocalVue()

Vue.use(VueI18n)
const i18n = new VueI18n({});

describe('HelpDef.vue', () => {
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  const mountFunction = options => {
    return mount(HelpDef, {
      localVue,
      vuetify,
      i18n,
      ...options,
    })
  }

  it('should render correct home form', () => {
    const wrapper = mountFunction()

    expect(wrapper.html()).toMatchSnapshot()
  })

})
