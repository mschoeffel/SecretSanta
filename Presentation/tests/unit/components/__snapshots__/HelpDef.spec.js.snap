// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpDef.vue should render correct home form 1`] = `
<v-container fluid="" class="fill-height">
  <v-row align="center" justify="center">
    <v-col cols="12" sm="12" md="8">
      <v-card class="elevation-12">
        <v-toolbar color="primary" dark="" flat="">
          <v-toolbar-title name="title" id="title">help.formtitle</v-toolbar-title>
        </v-toolbar>
        <v-card-text>
          <v-expansion-panels>
            <v-expansion-panel>
              <v-expansion-panel-header>help.firststeps</v-expansion-panel-header>
              <v-expansion-panel-content>help.firststeps-content</v-expansion-panel-content>
            </v-expansion-panel>
            <v-expansion-panel>
              <v-expansion-panel-header>help.newgroup</v-expansion-panel-header>
              <v-expansion-panel-content>help.newgroup-content</v-expansion-panel-content>
            </v-expansion-panel>
            <v-expansion-panel>
              <v-expansion-panel-header>help.drawmember</v-expansion-panel-header>
              <v-expansion-panel-content>help.drawmember-content</v-expansion-panel-content>
            </v-expansion-panel>
          </v-expansion-panels>
        </v-card-text>
        <v-card-actions>
          <v-btn name="routehome" id="routehome" color="secondary">help.home</v-btn>
        </v-card-actions>
      </v-card>
    </v-col>
  </v-row>
</v-container>
`;
